pkg/hubble/parser/seven/http.go

```diff
@@ -7,15 +7,26 @@ import (
 	"fmt"
 	"net/url"
 	"sort"
+	"strings"
 	"time"
 
+	"golang.org/x/exp/slices"
+
 	flowpb "github.com/cilium/cilium/api/v1/flow"
 	"github.com/cilium/cilium/pkg/hubble/defaults"
 	"github.com/cilium/cilium/pkg/hubble/parser/options"
 	"github.com/cilium/cilium/pkg/proxy/accesslog"
 )
 
 func decodeHTTP(flowType accesslog.FlowType, http *accesslog.LogRecordHTTP, opts *options.Options) *flowpb.Layer7_Http {
+	allowedHeaders := []string{
+		"traceparent",
+		"tracestate",
+		"cache-control",
+		"content-encoding",
+		"accept",
```

Comment:
What's the reasoning behind this list? In my opinion it should be longer, otherwise I foresee multiple requests to add a particular header that someone needs.

I'm not sure what's the best source for those, I would go through the [Wikipedia list](https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Standard_request_fields) and allow standard (non-deprecated) and common headers that are not sensitive. But maybe there is a better list somewhere. Some Envoy headers might be needed too.

In addition to `traceparent` and `tracecontext`, Hubble should allow headers for other common propagation formats: [baggage](https://www.w3.org/TR/baggage/#baggage-http-header-format), [X-B3-*](https://github.com/openzipkin/b3-propagation), [X-Datadog-*](https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces/?tab=datadog#how-are-rum-resources-linked-to-traces), [X-Amzn-Trace-Id](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-request-tracing.html). Hubble doesn't parse these headers itself, but they are used to correlate traces with flows in external systems.